src/app/persister.js

```diff
@@ -0,0 +1,25 @@
+import { get, set, del } from 'idb-keyval'
+import throttle from 'lodash.throttle'
+
+/**
+ * Creates an Indexed DB persister
+ * @see https://react-query.tanstack.com/plugins/persistQueryClient#building-a-persistor
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API
+ */
+
+const createIDBPersister = (idbValidKey = 'reactQuery') => {
+    // Throttle persisting by a second
+    const persistClient = throttle((client) => set(idbValidKey, client), 1000)
```

Comment:
Could you add an explanation for why this is necessary?

Reply:
Added comment in adf5f2846b05022886718924eecdd31e46b8800c. This is just to prevent writing to the cache more than once per second. When multiple queries are fetching you could potentially get a ton of calls to the persist fn. Which seems unnecessarily taxing. With this throttle we'll write on the leading and trailing edge so we should always write the last call to persist to the indexeddb.

Worst case scenario, the app crashes within the timeout of a second and less than a second of data could be lost. That seems like an ok tradeoff to me.